src/btree/bt_read.c

```diff
@@ -188,6 +188,10 @@ __las_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
 		total_incr += incr;
 		upd->txnid = las_txnid;
 		upd->prepare_state = prepare_state;
+		// If it isn't already done, set that we found prepared updates.
+		if (preparedp &&
```

Comment:
is there a case where preparedp can be NULL ?

Reply:
Yes, if a caller doesn't need and pass it as NULL. I have tried to code this function as a utility, if the caller doesn't care whether there are prepared updates or not, it can chose to pass as NULL. That said, there are not such callers at the moment.

Reply:
I think we can remove this check. Any way it is not a big structure where caller can pass NULL and save some memory, it is just a bool, caller can always pass the pointer and ignore the return value if not bothered.

Reply:
Is there any reason you feel like we should not check that passed variable is null ?

Doing so offers the benefit of not relying on the caller passing a variable to us.
It is also beneficial for the caller, if it doesn't need to know about the prepared state, it can just pass null, rather than going through the exercise of allocating a variable and then calling WT_UNUSED(..) on it.

I do not see any cons of doing a NULL check, while it offers the above said benefits.
__wt_log_release() for instance does something similar.

If you do not have strong reasons to remove this check I will like to keep it.